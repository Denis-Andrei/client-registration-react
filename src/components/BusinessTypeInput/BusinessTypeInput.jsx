import React, {useState} from 'react'
import { BackLink, H1, Label, ErrorText, FormGroup, Radio } from 'govuk-react'
import { Button } from 'govuk-react'
import { useHistory } from 'react-router-dom';
import { UserProperties } from '../../utils/UserProperties';
import { getData } from '../../utils/Utils';

export default function BusinessTypeInput() {
    const history = useHistory(); 
    const [businessType, setBusinessType] = useState(getData(UserProperties.businessType))
    const [hasErrors, setHasErrors] = useState(false)

    const handleChange = (e) => {
        console.log(e.target.value)
        setBusinessType(e.target.value)
    }

    const handleSubmit = () => {
        if(businessType){
            sessionStorage.setItem(UserProperties.businessType, businessType)
            history.push('/password')
        }else{
            setHasErrors(true)    
        }
    }

    return (
        <div>
            <BackLink data-testid="goBack" onClick={ () => history.goBack()}>Back</BackLink>
            <H1 data-testid="heading">Please select your business type</H1>
            <FormGroup data-testid="input-business-type" value={businessType} onChange={handleChange}>
                <Label>
                    {
                        hasErrors 
                        ? 
                        <ErrorText>Please select your business type</ErrorText>
                        :
                        ""
                    }
                    <Radio name="group1" value='Sole trader' defaultChecked={businessType === 'Sole trader'}>Sole trader.</Radio>
                    <Radio name="group1" value='Partnership' defaultChecked={businessType === 'Partnership'}>Partnership.</Radio>
                    <Radio name="group1" value='Private Limited' defaultChecked={businessType === 'Private Limited'}>Private Limited.</Radio>
                    <Radio name="group1" value='Public Limited' defaultChecked={businessType === 'Public Limited'}>Public Limited.</Radio>
                </Label>
            </FormGroup>
            <Button data-testid="submit" onClick={handleSubmit}>Continue</Button>
        </div>
    )
}
