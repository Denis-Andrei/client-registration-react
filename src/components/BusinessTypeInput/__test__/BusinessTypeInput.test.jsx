import React from 'react';
import ReactDOM from 'react-dom';
import BusinessTypeInput from '../BusinessTypeInput';
import { render, cleanup, fireEvent } from '@testing-library/react'
import renderer from "react-test-renderer"



afterEach(cleanup);

//Mocking history.push() and history.goBack()
const mockHistoryPush = jest.fn();
const mockGoBack = jest.fn();

    jest.mock('react-router-dom', () => ({
    useHistory: () => ({
        push: mockHistoryPush,
        goBack: mockGoBack
    }),
}));
//end of mockingw

it("BusinessTypeInput() renders withoud crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<BusinessTypeInput />, div)
})

it("renders BusinessTypeInput() correctly", () => {
    const { getByTestId }  = render(<BusinessTypeInput />)
    expect(getByTestId('heading')).toHaveTextContent("Please select your business type")
})

it("matches snapshot", () => {
    const tree = renderer.create(<BusinessTypeInput />).toJSON();
    expect(tree).toMatchSnapshot();
})
// describe("handleChange() ", () => {
//     it("updates businessName input on change event", () => {
//         const { getByTestId } = render(<BusinessTypeInput />)
//         const businessTypeInput = getByTestId("input-business-type");
    
//         fireEvent.change(businessTypeInput, {target: {value: "Sole trader"}})
//         expect(businessTypeInput.value).toBe("Sole trader")
//     })
// })

// describe("handleSubmit() ", () => {
//     it("onSubmit redirects to the correct route", () => {

    

//         const { getByTestId } = render(<BusinessTypeInput />)
//         const submit = getByTestId("submit");

//         fireEvent.click(submit);
//         expect(mockHistoryPush).toHaveBeenCalledWith('/password');
//     })
// })

describe("goBack() ", () => {
    it("should be called on click event", () => {
        const { getByTestId } = render(<BusinessTypeInput />)
        const goBackBtn = getByTestId("goBack");
    
        fireEvent.click(goBackBtn)
        expect(mockGoBack).toHaveBeenCalled();
    })
})