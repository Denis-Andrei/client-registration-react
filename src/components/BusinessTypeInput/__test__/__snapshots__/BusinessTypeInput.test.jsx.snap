// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <a
    className="src__Anchor-sc-11edswm-0 gGueb"
    data-testid="goBack"
    onClick={[Function]}
  >
    Back
  </a>
  <h1
    className="heading__StyledHeading-sc-1sovo83-0 juamsS"
    data-testid="heading"
    size="XLARGE"
  >
    Please select your business type
  </h1>
  <div
    className="src__FormGroup-kqurat-0 fREKLF"
    data-testid="input-business-type"
    onChange={[Function]}
    value=""
  >
    <label
      className="src__StyledLabel-sc-14n2scx-0 kHdLXy"
    >
      
      <label
        className="src__Label-sc-7h7h1q-0 bJdWRC"
      >
        <input
          className="src__Input-sc-7h7h1q-1 cVoNtu"
          defaultChecked={false}
          name="group1"
          type="radio"
          value="Sole trader"
        />
        <span
          className="src__LabelText-sc-7h7h1q-2 hVPuLQ"
        >
          Sole trader.
        </span>
      </label>
      <label
        className="src__Label-sc-7h7h1q-0 bJdWRC"
      >
        <input
          className="src__Input-sc-7h7h1q-1 cVoNtu"
          defaultChecked={false}
          name="group1"
          type="radio"
          value="Partnership"
        />
        <span
          className="src__LabelText-sc-7h7h1q-2 hVPuLQ"
        >
          Partnership.
        </span>
      </label>
      <label
        className="src__Label-sc-7h7h1q-0 bJdWRC"
      >
        <input
          className="src__Input-sc-7h7h1q-1 cVoNtu"
          defaultChecked={false}
          name="group1"
          type="radio"
          value="Private Limited"
        />
        <span
          className="src__LabelText-sc-7h7h1q-2 hVPuLQ"
        >
          Private Limited.
        </span>
      </label>
      <label
        className="src__Label-sc-7h7h1q-0 bJdWRC"
      >
        <input
          className="src__Input-sc-7h7h1q-1 cVoNtu"
          defaultChecked={false}
          name="group1"
          type="radio"
          value="Public Limited"
        />
        <span
          className="src__LabelText-sc-7h7h1q-2 hVPuLQ"
        >
          Public Limited.
        </span>
      </label>
    </label>
  </div>
  <button
    className="sc-bdnxRM gXVtdA"
    data-testid="submit"
    disabled={false}
    onClick={[Function]}
  >
    Continue
  </button>
</div>
`;
